src/adapter/http/src/data/metadata_handler.rs

```diff
@@ -79,34 +80,41 @@ impl RequestParams {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#[derive(Debug, serde::Serialize)]
+#[derive(Debug, serde::Serialize, utoipa::ToSchema)]
+#[schema(as = DatasetMetadataResponse)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
-pub struct ResponseBody {
+pub struct Response {
```

Comment:
Why not to name the structure identically to schema?

Reply:
I was pursuing a structure like:
```rust
mod handler_a {
  struct Request { .. }
  struct Response { .. }
  async fn handler_a() { .. }
}
mod handler_b {
  struct Request { .. }
  struct Response { .. }
  async fn handler_b() { .. }
}
```
since all these handlers and structs end up in a shared namespace of OpenAPI spec and have to be renamed anyway I guess this ruins this modularization idea.

I will change to unique struct names for now and we can revisit later.